mapserver/django/django_project/fba/management/commands/ingest_boundary_data.py

```diff
@@ -28,88 +30,168 @@ def is_table_exists(self, conn, table_name, schema_name):
+                sql = f'SELECT * FROM {schema_name}.{table_name}'
+                cur.execute(sql)
+                return cur.fetchall()
+        except:
+            return []
 
     def create_connection(self):
         try:
-            conn = driv.connect(database='gis',
-                                user='docker',
-                                password='docker',
-                                host='192.168.1.197',
-                                port='45432')
+            conn = driv.connect(database='geocris',
+                                user='cdema',
+                                password='cdema',
+                                host='192.168.100.6',
+                                port='35432')
             register(conn)
             return conn
         except driv.OperationalError as e:
             print(e)
         return None
 
+    def get_all_tables(self, conn, schema):
+        query = f"""
+            SELECT table_name FROM information_schema.tables
+            WHERE table_schema = '{schema}' AND table_type = 'BASE TABLE'
+            """
+        with conn.cursor() as cur:
+            cur.execute(query)
+            return [r[0] for r in cur.fetchall()]
+
     def get_all_schemas(self, conn):
-        cur = conn.cursor()
-        query = (
-            'SELECT schema_name FROM information_schema.schemata '
-            'WHERE length(schema_name) = 3'
-        )
-        cur.execute(query)
-        schemas_list = cur.fetchall()
-        schema_names = []
-        for schema_name in schemas_list:
-            schema_names.append(schema_name[0])
-        cur.close()
-        return schema_names
+        try:
+            with conn.cursor() as cur:
+                query = (
+                    'SELECT schema_name FROM information_schema.schemata '
+                    'WHERE length(schema_name) = 3'
+                )
+                cur.execute(query)
+                schemas_list = cur.fetchall()
+                return [r[0] for r in schemas_list]
+        except:
+            return []
+
+    def generate_admin_code(self, schema_name, adm_level, origin_id):
+        partition_names = ['country', 'district', 'sub_district', 'village']
+        partition_name = partition_names[adm_level]
+
+        try:
+            last_obj = AdminBoundaryGlobalKeyMapping.objects.filter(
+                partition_level=adm_level).order_by('-id_mapping').first()
+            last_id = last_obj.id_mapping + 1
+        except:
+            last_id = 1
+
+        admin_code_name = f'{schema_name}:{adm_level}:{origin_id}'
+
+        data = {
+            'key': admin_code_name,
+            'partition_level': adm_level,
+            'defaults': {
+                'id_mapping': last_id,
+                'partition': partition_name,
+            }
+        }
+
+        key_mapping, created = AdminBoundaryGlobalKeyMapping.objects.get_or_create(
+            **data)
+
+        return key_mapping.id_mapping
```

Comment:
Code to create newly discovered mapping between original row source (combination of schema name, table id, and administrative level) and the administrative code used inside FBA, because there are no official global distinct admin code at the moment within CDEMA countries. 

The mapping should stay the same for subsequent run. If there are new data, then the code generate new mapping.